frontend/src/components/common/FridgeCheckInDescription.tsx

```diff
@@ -1,36 +1,53 @@
 import { EditIcon } from "@chakra-ui/icons";
 import { Button, Link, Text } from "@chakra-ui/react";
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
 
-const FridgeCheckInDescription = () => (
-  <>
-    <Text textStyle={["mobileHeader4", "desktopSubtitle"]} pt="2rem">
-      Fridge check-in description
-      <Button
-        variant="editInfo"
-        rightIcon={<EditIcon size={24} />}
-        onClick={() => alert("Edit content clicked!")}
-      />
-    </Text>
-    <Text textStyle={["mobileBody", "desktopBody"]} pt="2rem">
-      Select a card to see more details pertaining to the upcoming donation.
-      Select a card to see more details pertaining to the upcoming donation.
-      Select select a card to see more details pertaining to the Select a card
-      upcoming donation. Select a card to see more details pertaining to the
-      upcoming donation. Select a card to see more details pertaining to the
-      upcoming donation. Select a card Select a card to see more details
-      pertaining to the upcoming donation. Select a card to see more details
-      pertaining to the upcoming donation.
-      <Link
-        color="#498FB6"
-        textDecoration="underline"
-        textStyle={["mobileLink", "desktopLink"]}
-        href="www.google.com"
-        isExternal
-      >
-        Link to instructions
-      </Link>
-    </Text>
-  </>
-);
+import ContentAPIClient from "../../APIClients/ContentAPIClient";
+import * as Routes from "../../constants/Routes";
+import { Content } from "../../types/ContentTypes";
+
+const FridgeCheckInDescription = () => {
+  const [content, setContent] = useState<Content>();
+
+  const history = useHistory();
+  const navigateToEditPage = () => {
+    history.push(Routes.ADMIN_CHECK_IN_EDIT_DESCRIPTION_PAGE);
+  };
+
+  useEffect(() => {
+    const getContent = async () => {
+      const contentResponse = await ContentAPIClient.getContent();
+      setContent(contentResponse);
+    };
+
+    getContent();
+  }, []);
+
+  return (
+    <>
+      <Text textStyle={["mobileHeader4", "desktopSubtitle"]} pt="2rem">
+        Fridge check-in description
+        <Button
+          variant="editInfo"
+          rightIcon={<EditIcon size={24} />}
+          onClick={navigateToEditPage}
+        />
+      </Text>
+      <Text textStyle={["mobileBody", "desktopBody"]} pt="2rem">
+        {content?.checkinDescription}
+        <br />
+        <br />
+        <Link
+          color="#498FB6"
+          textStyle={["mobileLink", "desktopLink"]}
+          href={content?.checkinUrl}
+          isExternal
+        >
+          Link to instructions
+        </Link>
```

Comment:
can we do this for both of the content components so that if a link shows up it will be existant?
```suggestion
        { content.checkInUrl && <Link
          color="#498FB6"
          textStyle={["mobileLink", "desktopLink"]}
          href={content.checkinUrl}
          isExternal
        >
          Link to instructions
        </Link>}
```